### PR TITLE
Update CAFE to 5.1

### PR DESCRIPTION
**Pull request by benfulton**
Update CAFE to 5.1

### CODE REVIEW

File: recipes/cafe/meta.yaml [hunk 1]
```diff
@@ -10,8 +10,10 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 3
+  number: 1
```
Comment (thanhleviet) [inline recipes/cafe/meta.yaml:13]: should be reset to 0 as this is a new version

File: recipes/cafe/meta.yaml [hunk 2]
```diff
@@ -10,8 +10,10 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 3
+  number: 0
   skip: True # [osx]
+  run_exports:
+    - {{ pin_subpackage('cafe', max_pin="x") }}
```
Comment (thanhleviet) [inline recipes/cafe/meta.yaml:16]: `- {{ pin_subpackage('cafe', max_pin="x.x") }}`

**Comment by thanhleviet** [comment]
It passed linting but testing failed due to 404 (https://github.com/hahnlab/CAFE5/releases/download/v5.0/CAFE5-5.1.0.tar.gz)

**Comment by benfulton** [comment]
@BiocondaBot please add label

**Review by mencian — APPROVED** [review]
Thanks, @benfulton!

**Pull request merged**